libscript/sources/Parser.cxx

```diff
@@ -548,6 +635,25 @@ void Parser::consumeSystemStatement(Node &result, Token &lastToken)
     }
     result.children.push_back(statementResult);
 }
+void Parser::consumeEmitStatement(Node &result, [[maybe_unused]] Token &lastToken)
+{
+    Node emitEvent = {
+        .type = NodeType::EmitEvent,
```

Comment:
Even if front() does return a ref, it could be copied if you don't mark the ref on the auto

Reply:
The line is not the correct one. Look at line 209

Reply:
It was on purpose. The code segfaults otherwise, as the call to ` consumeSystemFuncParams(emitEvent, eventToken)` takes a dangling reference after `_tokens.pop()` is called.